### PR TITLE
Add io.github.plrigaux.sysd-manager

### PR DESCRIPTION
**Pull request by plrigaux**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.

  This application exists to easily allow user to manage their **systemd units** via a GUI. You are able to make changes to the enablement and running status of each of the units, also you are able to view  their unit files. 

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
I don't understand why you close the original PR, but this https://github.com/flathub/flathub/pull/5468#pullrequestreview-2216586740

Was never addressed.

**Comment by plrigaux** [comment]
I close the original pr because I deleted my repo, because I wasn't able to
push anything on it.

I don't understand what the issue about #5468.

What does mean: Flathub submission should not contain the upstream source
code.

Please just tell me what do to fix it

Thanks

Pierre-Luc Rigaux
514 808 6241

**Comment by plrigaux** [comment]
Ok I made changes. I removed all source files and took a look to https://docs.flathub.org/docs/for-app-authors/requirements#required-files

Tell me if everything ok?

Thanks

File: io.github.plrigaux.sysd-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+id: io.github.plrigaux.sysd-manager
+runtime: org.gnome.Platform
+runtime-version: '46'
+sdk: org.gnome.Sdk
+#runtime: org.freedesktop.Platform
+#runtime-version: '23.08'
+#sdk: org.freedesktop.Sdk
+sdk-extensions:
+- org.freedesktop.Sdk.Extension.rust-stable
+- org.freedesktop.Sdk.Extension.llvm16
```
Comment (bbhtt) [inline io.github.plrigaux.sysd-manager.yaml:10]:
```suggestion
- org.freedesktop.Sdk.Extension.llvm18
```

File: io.github.plrigaux.sysd-manager.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+id: io.github.plrigaux.sysd-manager
+runtime: org.gnome.Platform
+runtime-version: '46'
+sdk: org.gnome.Sdk
+#runtime: org.freedesktop.Platform
+#runtime-version: '23.08'
+#sdk: org.freedesktop.Sdk
```
Comment (bbhtt) [inline io.github.plrigaux.sysd-manager.yaml:7]:
```suggestion
```

File: io.github.plrigaux.sysd-manager.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  - install -Dm644 ./data/icons/hicolor/scalable/apps/io.github.plrigaux.sysd-manager.svg -t /app/share/icons/hicolor/scalable/apps/
+  - install -Dm644 ./data/schemas/io.github.plrigaux.sysd-manager.gschema.xml -t /app/share/glib-2.0/schemas
+  - install -Dm644 ./data/metainfo/io.github.plrigaux.sysd-manager.metainfo.xml -t /app/share/metainfo
+  - install -Dm644 ./screenshots/* -t /app/screenshots
+  - glib-compile-schemas /app/share/glib-2.0/schemas
+
+  sources:
+  - type: dir
+    path: "."
+  - "cargo-sources.json"
```
Comment (bbhtt) [inline io.github.plrigaux.sysd-manager.yaml:64]: dir source isn't going to work, ci does not have access to your PC.

File: io.github.plrigaux.sysd-manager.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    env:
+      CARGO_HOME: /run/build/sysd-manager/cargo
+  buildsystem: simple
+  build-options:
+    env:
+      CARGO_HOME: /run/build/sysd-manager/cargo
+  build-commands:
+  - cargo --offline fetch --manifest-path Cargo.toml --verbose
+  - cargo --offline build --release --verbose
+  - pwd
```
Comment (bbhtt) [inline io.github.plrigaux.sysd-manager.yaml:52]:
```suggestion

```

File: io.github.plrigaux.sysd-manager.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  - install -Dm755 ./target/release/sysd-manager -t /app/bin/
+  - install -Dm644 ./data/applications/io.github.plrigaux.sysd-manager.desktop -t /app/share/applications
+  - install -Dm644 ./data/icons/hicolor/scalable/apps/io.github.plrigaux.sysd-manager.svg -t /app/share/icons/hicolor/scalable/apps/
+  - install -Dm644 ./data/schemas/io.github.plrigaux.sysd-manager.gschema.xml -t /app/share/glib-2.0/schemas
+  - install -Dm644 ./data/metainfo/io.github.plrigaux.sysd-manager.metainfo.xml -t /app/share/metainfo
+  - install -Dm644 ./screenshots/* -t /app/screenshots
```
Comment (bbhtt) [inline io.github.plrigaux.sysd-manager.yaml:58]:
```suggestion
```

File: io.github.plrigaux.sysd-manager.yaml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+modules:
+- name: sysd-manager
+  buildsystem: simple
+  build-options:
+    env:
+      CARGO_HOME: /run/build/sysd-manager/cargo
+  buildsystem: simple
+  build-options:
+    env:
+      CARGO_HOME: /run/build/sysd-manager/cargo
```
Comment (bbhtt) [inline io.github.plrigaux.sysd-manager.yaml:48]:
```suggestion
```

duplicate

File: io.github.plrigaux.sysd-manager.yaml [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+- --env=SYSDMNG_DIST_MODE=flatpack
+# Typical GNOME and GTK applications should use:
+- --talk-name=org.gtk.vfs.*
+- --filesystem=xdg-run/gvfsd
+# Needed to talk to systemd1 with dbus
+- --talk-name=org.freedesktop.systemd1
+- --system-talk-name=org.freedesktop.systemd1
+
+build-options:
+  append-path: "/usr/lib/sdk/rust-stable/bin:/usr/lib/sdk/llvm16/bin"
```
Comment (bbhtt) [inline io.github.plrigaux.sysd-manager.yaml:32]:
```suggestion
  append-path: "/usr/lib/sdk/rust-stable/bin:/usr/lib/sdk/llvm18/bin"
```

File: io.github.plrigaux.sysd-manager.yaml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+- org.freedesktop.Sdk.Extension.llvm16
+command: sysd-manager
+finish-args:
+- --share=ipc
+- --socket=fallback-x11
+- --socket=wayland
+- --device=dri
+#- --filesystem=home
+- --filesystem=host
+#- --talk-name=org.freedesktop.Flatpak"
```
Comment (bbhtt) [inline io.github.plrigaux.sysd-manager.yaml:19]:
```suggestion
```

File: io.github.plrigaux.sysd-manager.yaml [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+sdk-extensions:
+- org.freedesktop.Sdk.Extension.rust-stable
+- org.freedesktop.Sdk.Extension.llvm16
+command: sysd-manager
+finish-args:
+- --share=ipc
+- --socket=fallback-x11
+- --socket=wayland
+- --device=dri
+#- --filesystem=home
```
Comment (bbhtt) [inline io.github.plrigaux.sysd-manager.yaml:17]:
```suggestion
```

File: io.github.plrigaux.sysd-manager.yaml [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+- --filesystem=host
```
Comment (bbhtt) [inline io.github.plrigaux.sysd-manager.yaml:18]: why is host needed ?

File: io.github.plrigaux.sysd-manager.yaml [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+#- --talk-name=org.freedesktop.Flatpak"
+- --env=RUST_LOG=info
+- --env=G_MESSAGES_DEBUG=none
```
Comment (bbhtt) [inline io.github.plrigaux.sysd-manager.yaml:21]:
```suggestion
```

File: io.github.plrigaux.sysd-manager.yaml [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+- --env=RUST_BACKTRACE=1
+- --env=SYSDMNG_DIST_MODE=flatpack
+# Typical GNOME and GTK applications should use:
+- --talk-name=org.gtk.vfs.*
+- --filesystem=xdg-run/gvfsd
+# Needed to talk to systemd1 with dbus
+- --talk-name=org.freedesktop.systemd1
+- --system-talk-name=org.freedesktop.systemd1
+
```
Comment (bbhtt) [inline io.github.plrigaux.sysd-manager.yaml:30]: why do you need this ?

File: io.github.plrigaux.sysd-manager.yaml [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+#- --filesystem=home
+- --filesystem=host
+#- --talk-name=org.freedesktop.Flatpak"
+- --env=RUST_LOG=info
+- --env=G_MESSAGES_DEBUG=none
+- --env=RUST_BACKTRACE=1
+- --env=SYSDMNG_DIST_MODE=flatpack
+# Typical GNOME and GTK applications should use:
+- --talk-name=org.gtk.vfs.*
+- --filesystem=xdg-run/gvfsd
```
Comment (bbhtt) [inline io.github.plrigaux.sysd-manager.yaml:26]: is this actually needed ?

**Comment by bbhtt** [comment]
PR checklist is not done. Can you complete that ?

File: io.github.plrigaux.sysd-manager.yaml (hunk 13 above)
Comment (plrigaux) [reply io.github.plrigaux.sysd-manager.yaml]: Nope
I checked and removed those lines

**Comment by hfiguiere** [comment]
> Please describe the application briefly.

Still not done

File: io.github.plrigaux.sysd-manager.yaml (hunk 12 above)
Comment (plrigaux) [reply io.github.plrigaux.sysd-manager.yaml]: The program is based on having dbus access to systemd. It does api calls as defined here https://www.freedesktop.org/wiki/Software/systemd/dbus/ to retrieve the list of units (services), start service, stop service, get unit parameters, ...

Basically the program is a GUI of "systemctl". Without those 2 access the program is meaningless.

Thank you

File: io.github.plrigaux.sysd-manager.yaml (hunk 10 above)
Comment (hfiguiere) [reply io.github.plrigaux.sysd-manager.yaml:14]: it doesn't have access to the host `/usr`. Did you test it?

**Comment by plrigaux** [comment]
> > Please describe the application briefly.
> 
> Still not done

now done

File: io.github.plrigaux.sysd-manager.yaml (hunk 10 above)
Comment (bbhtt) [reply io.github.plrigaux.sysd-manager.yaml:14]: You need to look at `/run/host` when inside the Flatpak.

Comment (bbhtt) [reply io.github.plrigaux.sysd-manager.yaml:14]: Do you need `home` too ? If not use `host-os` which has less access than plain `host`. see https://docs.flatpak.org/en/latest/sandbox-permissions.html#filesystem-access

**Comment by bbhtt** [comment]
bot, build io.github.plrigaux.sysd-manager

File: io.github.plrigaux.sysd-manager.yaml (hunk 10 above)
Comment (plrigaux) [reply io.github.plrigaux.sysd-manager.yaml:14]: > You need to look at `/run/host` when inside the Flatpak.

I know I tested it. When inside a flatpack the program point to /run/host/usr/lib/systemd/...

**Comment by plrigaux** [comment]
No it doesn't need to show system units, flatpak-spawn.

I will make modifications, but should I new to do a new PR of just commit a new git tag version in the manifest?

**Comment by plrigaux** [comment]
I removed the flatpak-spawn check

**Comment by bbhtt** [comment]
bot, build io.github.plrigaux.sysd-manager

**Pull request closed without merging**